src/test/java/com/softserve/academy/spaced/repetition/controller/CourseControllerTest.java

```diff
@@ -59,200 +70,96 @@ public void getCourseById() throws Exception {
-                        "      \"rating\": 0," +
-                        "      \"image\": \"http://localhost/api/service/image/18\"," +
-                        "      \"published\": true," +
-                        "      \"ownerId\": 1," +
-                        "      \"categoryId\": 10," +
-                        "      \"courseId\": 5," +
-                        "      \"links\":[{\"rel\":\"self\",\"href\":\"http://localhost/api/category/10/courses/5\"},{\"rel\":\"decks\",\"href\":\"http://localhost/api/category/10/courses/5/decks\"}]"+
-                        "    }," +
-                        "    {" +
-                        "      \"name\": \"PHP interview course\"," +
-                        "      \"description\": \"2 parts of java questions & answers\"," +
-                        "      \"rating\": 0," +
-                        "      \"image\": \"http://localhost/api/service/image/17\"," +
-                        "      \"published\": true," +
-                        "      \"ownerId\": 1," +
-                        "      \"categoryId\": 4," +
-                        "      \"courseId\": 4," +
-                        "      \"links\":[{\"rel\":\"self\",\"href\":\"http://localhost/api/category/4/courses/4\"},{\"rel\":\"decks\",\"href\":\"http://localhost/api/category/4/courses/4/decks\"}]"+
-                        "    }" +
-                        "  ]," +
-                        "  \"last\": true," +
-                        "  \"totalPages\": 1," +
-                        "  \"totalElements\": 5," +
-                        "  \"size\": 0," +
-                        "  \"number\": 0," +
-                        "  \"sort\": null," +
-                        "  \"first\": true," +
-                        "  \"numberOfElements\": 5" +
-                        "}"));
+                .andExpect(jsonPath("$.size", is(QUANTITY_COURSES_IN_PAGE)))
+                .andExpect(jsonPath("$.sort[:1].ascending", hasItem(ASCENDING)))
+                .andExpect(jsonPath("$.sort[:1].descending", hasItem(!ASCENDING)))
+                .andExpect(jsonPath("$.sort[:1].property", hasItem(SORT_BY)));
     }
 
     private Page<Course> createCourses() throws ParseException {
         List<Course> courseList = new ArrayList<>();
 
-        Course course = new Course();
-        course.setId(3L);
-        course.setName("C# interview course");
-        course.setDescription("questions & answers");
-        course.setRating(0);
-        course.setImage(new Image(16L));
-        course.setPublished(true);
-        User user = new User();
-        Account account = new Account();
-        account.setId(1L);
-        user.setId(1L);
-        user.setAccount(account);
-        course.setOwner(user);
-        Category category = new Category();
-        category.setId(3L);
-        course.setCategory(category);
+        Course course = createCourse(3L, "C# interview course", "questions & answers",
```

Comment:
Don't you want change it to something like (not exactly):
```
for (int i = 0; i < 10; i++) {
    courseList.add(createCourse(i, "C++ interview course" + i, "3 parts of java questions & answers" + i, 0, 15L + i, true, i, "admin@gmail.com", 1, 2L));
}
```

Reply:
Is this obligatory?

Reply:
No, there is nothing obligatory. The question is do you want to learn any best paractices.